fvtest/compilertriltest/TernaryTest.cpp

```diff
@@ -0,0 +1,490 @@
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include "OpCodeTest.hpp"
+#include "default_compiler.hpp"
+#include "omrformatconsts.h"
+
+// C++11 upgrade (Issue #1916).
+template <typename CompareType, typename ValType>
+class TernaryTest : public TRTest::JitTest, public ::testing::WithParamInterface<std::tuple<std::tuple<CompareType, CompareType>, std::tuple<ValType, ValType>, ValType (*)(CompareType, CompareType, ValType, ValType)>> {};
+
+/** \brief
+ *     Struct equivalent of the TernaryTestParamStruct tuple.
+ *
+ *  \tparam CompareType
+ *     The type of the compare child of the ternary opcode.
+ *
+ *  \tparam ValType
+ *     The type that the ternary opcode returns.
+ */
+template <typename CompareType, typename ValType>
+struct TernaryTestParamStruct
+   {
+   CompareType c1;
+   CompareType c2;
+   ValType v1;
+   ValType v2;
+   ValType (*oracle)(CompareType, CompareType, ValType, ValType);
+   };
+
+/** \brief
+ *     Given an instance of TernaryTestParamStruct, returns an equivalent instance of TernaryTestParamStruct.
+ */
+template <typename CompareType, typename ValType>
+TernaryTestParamStruct<CompareType, ValType> to_struct(std::tuple<std::tuple<CompareType, CompareType>, std::tuple<ValType, ValType>, ValType (*)(CompareType, CompareType, ValType, ValType)> param)
+   {
+   TernaryTestParamStruct<CompareType, ValType> s;
+
+   s.c1 = std::get<0>(std::get<0>(param));
+   s.c2 = std::get<1>(std::get<0>(param));
+   s.v1 = std::get<0>(std::get<1>(param));
+   s.v2 = std::get<1>(std::get<1>(param));
+   s.oracle = std::get<2>(param);
+   return s;
+   }
+
+/** \brief
+ *     The oracle function which computes the expected result of the ternary test.
+ *
+ *  \tparam CompareType
+ *     The type of the values compared by oracle function.
+ *  \tparam ValType
+ *     The return type of the oracle function.
+ */
+template <typename CompareType, typename ValType>
+static ValType xternaryOracle(CompareType c1, CompareType c2, ValType v1, ValType v2)
+   {
+   return ((c1 < c2) ? v1 : v2);
+   }
+
+/** \brief
+ *     The function computes the input vector of pairs for the compare node.
+ */
+template <typename T>
+static std::vector<std::pair<T, T>> compareInputs<T>()
```

Comment:
The `<T>` after `compareInputs` should not be there
```suggestion
static std::vector<std::pair<T, T>> compareInputs()
```

Reply:
@yanluo7 I believe the change I requested above will fix the build failures.

Reply:
> @yanluo7 I believe the change I requested above will fix the build failures.

Thanks, should be fixed.